packages/next/src/build/swc/options.ts

```diff
@@ -423,7 +440,7 @@ export function getLoaderSWCOptions({
               type: 'commonjs',
             },
           }
-        : {}),
+        : getModuleOptions(!!esm)),
```

Comment:
Better to use explicit type like:

```
esm: boolean | undefined 
```

and then add a default (`= false`) when destructuring or in the return than to cast between undefined/boolean.